src/deque.rs

```diff
@@ -1703,4 +1886,68 @@ mod tests {
         assert_eq!(deque.len(), 0);
         assert!(deque.is_empty());
     }
+
+    // Based on alloc's own VecDeque test
+    //
+    // https://github.com/rust-lang/rust/blob/fa3155a644dd62e865825087b403646be01d4cef/library/alloc/src/collections/vec_deque/tests.rs#L1086
+    //
+    // Tests that each element's destructor is called when being truncated.
```

Comment:
Please test the following cases:

- Truncating an empty deque
- Truncating an continuous deque
  - leaving it unchanged
  - leaving it empty
  - leaving it half-empty
- Truncating a discontinuous deque
  - leaving it unchanged
  - leaving it discontinuous but dropping some elements
  - leaving it continuous (i.e. all the back is dropped)
    - reducing the front slice
    - not touching the front slice
  - leaving it empty

Essentially ensure that all branches of the `truncate` function are exercized.

Reply:
Added more tests, thanks for the suggestions! I was surprised there wasn't more in the std, maybe I just didn't look deep enough. 

Let me know if any should be adjusted.